### PR TITLE
HK: don't progression balance "Currency"-like progression items

### PR DESCRIPTION
**Pull request by Berserker66**
fixes Ijwu/Archipelago.HollowKnight#46

### CODE REVIEW

File: worlds/hk/__init__.py [hunk 1]
```diff
@@ -321,6 +321,9 @@ def __init__(self, name, advancement, code, type, player: int = None):
         if name == "Mimic_Grub":
             self.trap = True
 
+        if type in ("Grub", "Charm", "DreamWarrior", "Root", "Egg"):
```
Comment (Ijwu) [inline worlds/hk/__init__.py:324]: Some charms are advancement items and should be balanced, though. Kono gave a list in the GH issue on my page. I can get you the charm IDs if that's what you need.

**Review by Ijwu — CHANGES_REQUESTED** [review]

File: worlds/hk/__init__.py (hunk 1 above)
Comment (Berserker66) [reply worlds/hk/__init__.py:324]: If not, it's actually names, not IDs

Comment (Ijwu) [reply worlds/hk/__init__.py:324]: Here's the list of names: https://github.com/Ijwu/Archipelago.HollowKnight/issues/46

Comment (KonoTyran) [reply worlds/hk/__init__.py]: charm list added in 6a715ff. I pulled the names directly out of ExtractedData.py if that is not correct then.. whooops

Comment (Ijwu) [reply worlds/hk/__init__.py]: Once berserker approves that commit, I'll approve the PR.

Comment (Ijwu) [reply worlds/hk/__init__.py]: Berserker says "yep looks good" in voice chat.

**Review by Ijwu — APPROVED** [review]

**Pull request merged**